### PR TITLE
Update loaded chains and apply initial covenants on deploy

### LINKED ISSUE #496: When a node is deployed, compare existing root manifest with new manifest and apply CLI changes
The deploying node should compare the current manifest with the new manifest and apply applicable CLI level udpates:

- destroy chains one time on start
- create new chains and configure them one time on start (including applying their covenant if they are brand new chains)

### PR DESCRIPTION
**Pull request by nicholeous**
Closes #496 

- initialise covenants on new chains by applying configured covenant
- destroy any chains that are no longer configured in the manifest if available to this cli

### CODE REVIEW

File: packages/interbit/src/chainManagement/destroyRemovedChains.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  const chainAliasesConfiguredOnRoot = getChainAliases(rootChainManifest)
+  const chainEntries = Object.entries(getChains(manifest))
+  const removedChains = chainEntries.filter(
+    chainEntry => !chainAliasesConfiguredOnRoot.includes(chainEntry[0])
+  )
+
+  for (const removedChain of removedChains) {
+    const removedChainInterface = cli.getChain(removedChain)
```
Comment (djlmills) [inline packages/interbit/src/chainManagement/destroyRemovedChains.js]: Something is not quite right here. The values in `removedChains` returned from filtering `Object.entries`  are key/value pairs (you filtering them using the chain alias from `chainEntry[0]`). Then I would expect the corresponding expression for getting the chain to be `cli.getChain(removedChain[1])` as that would be the chain ID. Alternatively deconstruct the chain alias and chain ID in the right spots.

Please be careful when naming variables that contain key/value pairs (aka `entries`) so we don't fall into these traps. I am also not keen on the naming `removedChain` and `removedChains` as I really don't know what type of object is being referred to (elsewhere in the code base, this is the chain interface).

File: packages/interbit/src/chainManagement/destroyRemovedChains.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (removedChainInterface) {
+      const destroyAction = destroy()
+      removedChainInterface.dispatch(destroyAction)
+    }
```
Comment (djlmills) [inline packages/interbit/src/chainManagement/destroyRemovedChains.js]: Of course there are no guarantees this will actually work. We have zombie chains.

Comment (nicholeous) [reply packages/interbit/src/chainManagement/destroyRemovedChains.js]: Have we filed a bug in core-coordination?

File: packages/interbit/src/chainManagement/destroyRemovedChains.js (hunk 1 above)
Comment (nicholeous) [reply packages/interbit/src/chainManagement/destroyRemovedChains.js]: Good eye, this should be caught in integrations tests when all the open issues are finalised in the cascading-deploy. To be doubly sure, I have opened a ticket for a wide ranging integration test that incorporates destroy #658

**Pull request merged**